File: Chalkable.Common/Exceptions/ChalkableException.cs
using System;

namespace Chalkable.Common.Exceptions
{
    public class ChalkableException : Exception
    {
        public virtual string Title { get; private set; } 
        public ChalkableException() : base(ChlkResources.ERR_CHALKABLE_EXCEPTION)
        {
        }

        public ChalkableException(string message, string title = null) : base(message)
        {
            Title = title;
        }

        public ChalkableException(string message, Exception innerException, string title = null)
            : base(message, innerException)
        {
            Title = title;
        }
    }
}

File: Chalkable.Web/app/chlk/lib/exception/ChalkableException.js
NAMESPACE('chlk.lib.exception', function () {
    "use strict";

    /** @class chlk.lib.exception.ChalkableException */
    EXCEPTION(
        'ChalkableException', [
            function $(msg_, e_, title_) {
                BASE(msg_ || 'Chalkable exception', e_);
                this.title = title_;
            },
            String, 'title'
        ]);
});